src/analyzer.ts

```diff
@@ -186,11 +200,34 @@ export class StreamAnalyzer extends Transform {
   /**
    * Attempts to retreive document-order transitive dependencies for `url`.
    */
-  _getDependencies(url: string): Promise<DocumentDeps> {
-    let dir = path.posix.dirname(url);
+  async _getDependencies(url: string): Promise<DocumentDeps> {
+    const doc = await this.analyzer.analyzeRoot(url);
+    const scripts = new Set<string>();
+    const styles = new Set<string>();
+    const imports = new Set<string>();
+
+    for (const importDep of doc.getByKind('import')) {
+      const importUrl = importDep.url;
+      if (isDependencyExternal(importUrl)) {
+        logger.debug(`ignoring external dependency: ${importUrl}`);
+      } else if (importDep.type === 'html-script') {
+        scripts.add(importUrl);
+      } else if (importDep.type === 'html-style') {
+        styles.add(importUrl);
+      } else if (importDep.type === 'html-import') {
+        imports.add(importUrl);
+      } else {
+        logger.warn(`unexpected import type encountered: ${importDep.type}`);
+      }
+    }
 
-    return this.analyzer.metadataTree(url)
-        .then((tree) => getDependenciesFromDocument(tree, dir));
+    const deps = {
+      scripts: Array.from(scripts),
```

Comment:
This `DocumentDeps` is a deprecated type that we'll be able to get rid of in the new bundler logic. I kept the interface as-is for now so that we wouldn't have to refactor its current usage in `_addDependencies()`, `bundler.ts` & `service-worker.ts`. Is this worth adding a comment for to clarify?